### PR TITLE
Improve the student course overview cards

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

We want to display more information (visually) for the first pages students see.

### Description

The course icon is now displayed, alongside numbers for exercises, lectures, and exams. Those can be clicked on if the number is greater than 0 and takes you to the corresponding sub-page. The score has a visual graph and the exercise displayed as next exercise is always the one with the closest due date.

### Steps for Testing

Basically just looking at the student courses view:
- Is the exercise with the closest due date (in the next 7 days) displayed?
- Are the numbers for exercises, lectures, and exams correct?
- Is the score graph correct?
- Do you see any visual glitches?

### Test Coverage

`course-card.component.ts`: 100% (and testing of the exercise service method)

### Screenshots

grafik

### CODE REVIEW

**Review by katjanakosic — APPROVED** [review]
Tested on TS1. Looks good to me :)

**Review by NicolasUllmann — COMMENTED** [review]
Tested on Server 2. Everything worked for me.

**Review by willergeorg — COMMENTED** [review]
Can't see any deviations either. Looks fine.

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3. Looks good to me

**Review by nicolasruscher — APPROVED** [review]
Looked at the code, lgtm! 👍  neat changes!
